static/js/components/CouponNotificationDialog.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+import React from 'react';
+import Dialog from 'material-ui/Dialog';
+import Button from 'react-mdl/lib/Button';
+import {
+  COUPON_CONTENT_TYPE_PROGRAM,
+  COUPON_AMOUNT_TYPE_FIXED_DISCOUNT,
+  COUPON_AMOUNT_TYPE_PERCENT_DISCOUNT,
+} from '../constants';
+import type { Coupon } from '../flow/couponTypes';
+import type { AvailableProgram } from '../flow/enrollmentTypes';
+import type { Course } from '../flow/programTypes';
+
+type CouponNotification = {
+  coupon:               Coupon,
+  couponProgram:        ?AvailableProgram,
+  couponCourse:         ?Course,
+  open:                 boolean,
+  setDialogVisibility:  (v: boolean) => void,
+};
+
+const CouponNotificationDialog = (
+  { coupon, couponProgram, couponCourse, open, setDialogVisibility }: CouponNotification
+) => {
+  const {
+    amount,
+    amount_type: amountType,
+    content_type: contentType,
+    program_id: programId,
+    object_id: objectId,
+  } = coupon;
+  let programName;
+  if ( couponProgram ) {
+    programName = `the ${couponProgram.title} MicroMasters program`;
+  } else {
+    programName = `program ID ${programId}`;
+  }
+
+  let title, message;
+  if ( contentType === COUPON_CONTENT_TYPE_PROGRAM ) {
+    if ( amountType === COUPON_AMOUNT_TYPE_PERCENT_DISCOUNT ) {
+      title = `Coupon applied: ${amount.times(100)}% off each course!`;
+      message = <p key='message'>
+        This coupon gives <strong>a discount
+        of { amount.times(100).toString() }% off</strong> the price
+        of <strong>each</strong> course in { programName }.
+      </p>;
+    } else if ( amountType === COUPON_AMOUNT_TYPE_FIXED_DISCOUNT ) {
+      title = `Coupon applied: $${amount} off each course!`;
+      message = <p key='message'>
+        This coupon gives <strong>a discount
+        of ${ amount.toString() } off</strong> the price
+        of <strong>each</strong> course in { programName }.
+      </p>;
+    }
+  } else {
```

Comment:
Coupons for course runs are not handled. I was under the impression that we only really needed to handle coupons for programs for this pull request, and I was doing extra work by handling coupons for courses. If we need to handle course runs, as well, we'll need to figure out how to phrase the notification to make it clearer.

Can we merge this pull request without course run support for now, and add it in the future if necessary? This pull request is getting pretty big even without it.